Add 7.1 version constant to 7.x branch

jasontedor opened the pull request:
This commit adds the 7.1 version constant to the 7.x branch.

droberts195 committed:
Fix the clock resolution to millis in ScheduledEventTests (#38506)

The clock resolution changed from jdk8 to jdk10, hence the
test is passing in jdk8 but failing in jdk10.  Scheduled
events are serialised and deserialised with millisecond
precision, making test fail in jdk 10 and higher.

Fixes a problem introduced by #38415 and the fix is
identical to the one that was made in #38405.

jasontedor committed:
Add 7.1 version constant to 7.x branch

This commit adds the 7.1 version constant to the 7.x branch.

jasontedor commented:
This fails in `VersionCollection`: 

https://github.com/elastic/elasticsearch/blob/f8ed6c15c42a9cf990bc432d617e33eba3e218d4/buildSrc/src/main/java/org/elasticsearch/gradle/VersionCollection.java#L243

The assumptions about the unreleased versions is incorrect.

elasticmachine commented:
Pinging @elastic/es-core-infra

alpar-t committed:
Fix incorrect handling of version collections

atorok commented:
We need to make changes to account for the platform specific changes for bwc builds

alpar-t committed:
Fix bwcBuilds for versions newer than 7.0.0

dnhatn committed:
move old versions to unreleased

jasontedor committed:
Fix imports

jasontedor committed:
Remove bogus assertion

jasontedor committed:
Use typeless APIs

jasontedor committed:
Revert "Use typeless APIs"

This reverts commit 22eeb0c08599933c109cdff089c5f8363b2ff74c.

jasontedor committed:
Some full cluster restart tests passing

jasontedor committed:
More fixes

alpar-t committed:
Fixes for rolling upgrade

alpar-t committed:
Fix full cluster restart tests

atorok commented:
full cluster restart still fails with some seeds:
```
  2> REPRODUCE WITH: ./gradlew :qa:full-cluster-restart:v7.0.0#oldClusterTestRunner -Dtests.seed=988DCA3188D1E86E -Dtests.class=org.elasticsearch.upgrades.FullClusterRestartIT -Dtests.method="testRecovery" -Dtests.security.manager=true -Dtests.locale=en-FM -Dtests.timezone=Asia/Pontianak -Dcompiler.java=11 -Druntime.java=11
FAILURE 1.16s | FullClusterRestartIT.testRecovery <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: expected:<297> but was:<292>
   >    at __randomizedtesting.SeedInfo.seed([988DCA3188D1E86E:597DB39DA58122C9]:0)
   >    at org.elasticsearch.upgrades.FullClusterRestartIT.assertTotalHits(FullClusterRestartIT.java:667)
   >    at org.elasticsearch.upgrades.FullClusterRestartIT.testRecovery(FullClusterRestartIT.java:777)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   >    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   >    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
   >    at java.base/java.lang.Thread.run(Thread.java:834)

```

atorok commented:
@elasticmachine run elasticsearch-ci/default-distro

alpar-t committed:
Revert unintentional docker change

DaveCTurner committed:
Refresh before getting doc count

tvernum commented on test/framework/src/main/java/org/elasticsearch/upgrades/AbstractFullClusterRestartTestCase.java:
```diff
@@ -63,6 +63,10 @@ public final boolean isRunningAgainstOldCluster() {
 
+    public final boolean isRunningAgainstAncientCluster() {
+        return oldClusterVersion.before(Version.V_7_0_0);
+    }
```
I think this method should check `runningAgainstOldCluster` (if it's really about which cluster we're running against) or be called something more like `isOldClusterAncient` (if it's just check the version of the old cluster).

Otherwise the name implies that the behaviour is like `isRunningAgainstOldCluster`,  but it is not.

jasontedor replied:
Oops, good catch @tvernum.

jasontedor replied:
The reason this works most of the time is because most of the invocations to it are already guarded by `if (isRunningAgainstOldVersion())`.

tvernum commented on qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java:
```diff
@@ -1132,7 +1121,7 @@ private void checkSnapshot(String snapshotName, int count, Version tookOnVersion
         // In 7.0, type names are no longer returned by default in get index template requests.
         // We therefore use the deprecated typed APIs when running against the current version.
-        if (isRunningAgainstOldCluster() == false) {
+        if (isRunningAgainstAncientCluster() == false) {
```
Based on the behaviour of `isRunningAgainstOldCluster` (but see my comment there about the name vs behaviour), this check seems wrong.

If the old cluster is 7.0.0 then `isRunningAgainstAncientCluster` will be false, and `INCLUDE_TYPE_NAME_PARAMETER` will be true, but I think we only want that if old version is pre-7 and we're running on the new version.

Similarly if we're testing against `6.7.x` the parameter will be added regardless of which cluster we're running against (old or new), which is not how this test worked before.

But maybe I'm confused about why we want this parameter.

```suggestion
        if (isRunningAgainstOldCluster() == false && getOldClusterVersion().major < 7) {
```

tvernum commented on qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java:
```diff
@@ -1155,18 +1151,33 @@ private void checkSnapshot(String snapshotName, int count, Version tookOnVersion
             logger.debug("Indexing document [{}]", i);
-            Request createDocument = new Request("POST", "/" + index + "/doc/" + i);
+            Request createDocument = new Request("POST", "/" + index + "/" + type + "/" + i);
+            if (isRunningAgainstAncientCluster() == false) {
```
I think is going to get caught by `isRunningAgainstAncientCluster` not checking which cluster we run on.

jasontedor replied:
I think all invocations (on mobile, so only going from memory from late last night) of this method are run against the old cluster (by a guard higher up the stack). We should fix the method though.

tvernum reviewed (commented):
I think this is right, but I have a concern around `isRunningAgainstAncientCluster`

atorok commented on qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java:
```diff
@@ -863,7 +845,7 @@ public void testRecovery() throws Exception {
     public void testSnapshotRestore() throws IOException {
         int count;
-        if (isRunningAgainstOldCluster()) {
+        if (isRunningAgainstOldCluster() && getOldClusterVersion().major < 8) {
```
It won't for this branch, but as we bump the versions I think we could get there.

jasontedor committed:
Fix method

alpar-t committed:
Fix x-pack full cluster restart

alpar-t committed:
Completly fix x-pack full cluster restart

jkakavas committed:
Fix watcher tests

alpar-t committed:
Fix checkstyle

martijnvg committed:
fixed rolling upgrade tests

atorok commented:
@elasticmachine run elasticsearch-ci/1

tbrooks8 commented:
This finicky test (https://github.com/elastic/elasticsearch/issues/38489) failed on CI/1. I will push an `AwaitsFix` if more changes need to be made to this PR (based on other failures).

martijnvg committed:
fixed wrong comparison

Tim-Brooks committed:
Mute tests

Tim-Brooks committed:
Merge branch 'add-seven-dot-one' of github.com:jasontedor/elasticsearch into add-seven-dot-one

martijnvg committed:
added pre 7 version and post 7 version of the test

martijnvg committed:
Revert "added pre 7 version and post 7 version of the test"

This reverts commit e1ceb5305614a3507cf672ada47e74ec7de3f665.

martijnvg committed:
finally fixed rolling upgrade tests, by muteing a test,
whether `types` field should be used should be decided later,
also it looks like this test was intented to be muted.
('tests.rest.blacklist' system property was in the wrong place)
